src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -413,6 +414,42 @@ def connection_is_activated(self, connection_id):
 
         return connection_is_activated
 
+    def get_default_gateway_state(self):
+        """Gets information about the current default gateway route. Returns a dictionary
+        including the gateway address, the interface name associated with the address, and
+        the NetworkManager connection associated with that device.
+        """
+
+        # TODO: broadcast signal if:
+        #   the default gateway IP changes
+        #   the interface associated with the default gateway changes
+        #   the connection associated with the interface of the default gateway changes
+
+        # to get an interface name from the default gatway:
+        # with pyroute2.IPRoute() as ipr:
+        #   ipr.get_links()[ipr.get_default_routes()[0]['attrs'][3][1] - 1]['attrs'][0][1]
+
```

Comment:
I suspect most of the prior 10 lines were not intended to be left in the final code.